contracts/services/kyc/KYCVerifierV1.sol

```diff
@@ -10,34 +10,56 @@ import {
     DeploymentBlockNonInitializable
 } from "../../DeploymentBlockNonInitializable.sol";
 import {ERC165} from "@openzeppelin/contracts/utils/introspection/ERC165.sol";
+import {EIP712} from "@openzeppelin/contracts/utils/cryptography/EIP712.sol";
+import {ECDSA} from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
 
 /**
  * @title KYCVerifierV1
  * @author Decent Labs
- * @notice Mock implementation of KYC verification service
- * @dev This contract implements IKYCVerifierV1, providing a simple KYC
- * verification service that always returns true.
+ * @notice KYC verification service using EIP-712 signature verification
+ * @dev This contract implements IKYCVerifierV1, providing KYC verification
+ * through cryptographic signature verification.
  *
  * Implementation details:
- * - Mock implementation for testing and development
- * - Always returns true for any address verification
+ * - Uses EIP-712 structured data signing for verification
+ * - Requires signature from authorized verifier address
  * - Deployed as singleton service per chain
- * - Production implementations would integrate with real KYC providers
+ * - Supports operating contract-specific verification
  *
- * Production considerations:
- * - Real implementations would check on-chain attestations
- * - Could integrate with identity protocols or oracles
- * - May use merkle trees or signature verification
- * - Should implement proper access controls for updates
+ * Security considerations:
+ * - Verifier address is immutable and set at deployment
+ * - Uses ECDSA signature recovery for verification
+ * - EIP-712 prevents signature replay across different domains
+ * - Operating contract context prevents cross-contract signature reuse
  *
  * @custom:security-contact security@decentlabs.io
  */
 contract KYCVerifierV1 is
     IKYCVerifierV1,
     IVersion,
     DeploymentBlockNonInitializable,
-    ERC165
+    ERC165,
+    EIP712
 {
+    // ======================================================================
+    // STATE VARIABLES
+    // ======================================================================
+
+    address private immutable _verifier;
```

Comment:
Please fix the solhint warning on this line

Reply:
and considering that this is not an upgradeable contract, and there's no way to change the `_verifier`, what is the remediation plan if the private key Decent is using on its server gets leaked? What would that affect and how would we fix it?

Reply:
If the private key does get leaked, then the hacker can potentially generate signatures attesting to any address being KYC verified. They could potentially generate a bunch of these signatures for their own addresses and buy up a large portion of the public sale tokens. They would of course, still have to pay for them.

We could make `KYCVerifierV1` be `ownable` and the `owner` can update the `verifier` address

Reply:
I assume Decent would be the owner of that contract? Maybe this does make sense, since it is _our address_ which is the verifier anyway.

Reply:
Yeah, I could see an argument that the verifier private keys could be considered less secure since they will be managed by a server in some way. 

And the owner could be a more secure address, like a Decent multisig.

Reply:
Made `KYCVerifierV1` inherit from `Ownable2Step`, and added an `onlyOwner` `updateVerifier` function